### DISCUSSION TITLE
Drop 2.11 support for 3.0.0 series?

### DISCUSSION THREAD

**Discussion by Quafadas** [lihaoyi/upickle#451] [Q&A] [answered]
Scala JS recently dropped 2.11 support. 
https://github.com/scala-js/scala-js/issues/4759

Some of upickles extra module dependencies, have also started to drop 2.11 support. I had a look at upickles build file with the aim of publishing ujson-circe for scala 3 and doing some random dependency upgrades. I realized that task required a bunch of 2.11 exceptions to be made.

- circe ujson needs special versions for 2.11
- same for play ujson
- As scala js no longer supports 2.11, continued support for 2.11 implies using multiple scala js versions in the build. 

I would personally choose to ease the maintenance / transition into scala 3,  in preference to preserve backward compatibility for 2.11 - which has been largely dopped - but this is a question for the maintainers / community.

Views? If this is objectionable for whatever reason, it is not that important to me - it's rather intended to help the maintainers.

**Comment by lefou** [comment]
This is an valid point. I typically promote backward compatibility, but in this case I agree, Scala 2.11 is rather old and upicke 2.0 is still usable with 2.11. 

Upickle 3.0 is all about thinking forward, supporting Scala 3 and its newer features like enums. Dropping it makes sense.

**Reply by Quafadas** [reply]
I'll see if I can get the tests to pass and submit a pr...

**Reply by lefou** [reply]
Great! Much appreciated. We planned to release 3.0.0 on this Sunday.

**Comment by SethTisue** [comment]
dooooooo iiiiiiiit

**Comment by lihaoyi** [comment]
IIRC we actually dropped 2.11 a while back, but @lolgab added it back in https://github.com/com-lihaoyi/upickle/pull/344, so we should get his opinion before dropping it again :P 

I personally don't have a strong opinion either way

**Reply by lolgab** [reply]
At that time we needed it for Scala Native, and other downstream libraries were using older versions for Scala 2.11.
Today Scala Native supports all Scala versions and most libraries stopped supporting it. I'm ok removing it :)
Thank you for asking :)

**Comment by Quafadas** [comment]
@SethTisue @lefou  I added my voice to reporting Luann and Lesley. Hope it helps. 

> More dumb chores for OSS maintainers to have to waste time on

:-(, although in this case, at least a smile at the irony of reporting spam posts being rate limited...

**Comment by lefou** [comment, marked as answer]
upickle 3.0.0 no longer supports Scala 2.11. If you're on Scala 2.11, you can still use upickle 2.0.0. Most changes between 2.0.0 and 3.0.0 are related to better Scala 3 support, so you shouldn't miss anything.
